packages/block-library/src/block/edit.js

```diff
@@ -20,30 +27,32 @@ import ReusableBlockEditPanel from './edit-panel';
 import ReusableBlockIndicator from './indicator';
 
 class ReusableBlockEdit extends Component {
-	constructor( { reusableBlock } ) {
+	constructor( { reusableBlock, registry } ) {
 		super( ...arguments );
 
 		this.startEditing = this.startEditing.bind( this );
 		this.stopEditing = this.stopEditing.bind( this );
-		this.setAttributes = this.setAttributes.bind( this );
+		this.setBlocks = this.setBlocks.bind( this );
 		this.setTitle = this.setTitle.bind( this );
 		this.save = this.save.bind( this );
 
-		if ( reusableBlock && reusableBlock.isTemporary ) {
+		if ( reusableBlock ) {
 			// Start in edit mode when we're working with a newly created reusable block
 			this.state = {
-				isEditing: true,
+				isEditing: reusableBlock.isTemporary,
 				title: reusableBlock.title,
-				changedAttributes: {},
+				blocks: parse( reusableBlock.content ),
 			};
 		} else {
 			// Start in preview mode when we're working with an existing reusable block
 			this.state = {
 				isEditing: false,
 				title: null,
-				changedAttributes: null,
+				blocks: [],
 			};
 		}
+
+		this.registry = createRegistry( { 'core/block-editor': storeConfig }, registry );
```

Comment:
Working locally in a separate branch, I observe a few more issues with the cloned registry:

- The copied store will not include the [custom middlewares we apply](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/middlewares.js). As I remarked at #14513, we should probably plan to do away with those middlewares altogether.
- The copied registry will not (at least with the current implementation of parent registries) include the plugins used for the parent, so generator actions (controls plugin) will fail. It's unclear to me whether we should (a) inherit plugins from a parent registry or (b) force the implementer who creates a child registry to re-declare the plugins (i.e. include a few following `this.registry.use` here). If we choose to go the "B" route, it should also be noted that we couldn't pre-load the store configuration when calling `createRegistry`, as the control plugin is [implemented](https://github.com/WordPress/gutenberg/blob/27c0d00102341615df838d1df2a40f1e33ff7973/packages/data/src/plugins/controls/index.js#L17) to only handle controls for a store configuration _after_ it's been included as a plugin, not retroactively.

Reply:
This just reinforces my ideas that:

 - `createNamespaceStore` would be better if it's independent from the registry (no need for registry argument) 
 - `use` is not as useful as it seems and both the controls and persistence plugins should be bundled as opt-in features in the `createNamespaceStore`